app/jenkins-for-jira-ui/src/components/ConnectionPanel/ConnectionPanelMain.tsx

```diff
@@ -78,7 +80,7 @@ const ConnectionPanelMain = ({
 
 	let setUpGuideUpdateAvailableContent;
 
-	if (isLoading || isUpdatingServer) {
+	if (isLoading || (isUpdatingServer && uuidOfRefreshServer === jenkinsServer.uuid)) {
```

Comment:
Noticed refresh wasn't working anymore. Must've broken it at some point after the initial implementation. Tested and made sure it now works in both the pending and update available states (only places it's currently used) and that it only refreshes the panel for the server where the button is clicked.